src/renderer/components/node/NodeOutputs.tsx

```diff
@@ -1,17 +1,83 @@
+/* eslint-disable react/jsx-props-no-spreading */
+
 import { memo, useCallback } from 'react';
 import { useContext, useContextSelector } from 'use-context-selector';
-import { Output, OutputId, SchemaId } from '../../../common/common-types';
+import { Output, OutputId, OutputKind, SchemaId } from '../../../common/common-types';
+import { Type } from '../../../common/types/types';
+import { assertNever } from '../../../common/util';
 import { ExecutionContext } from '../../contexts/ExecutionContext';
 import { GlobalContext } from '../../contexts/GlobalNodeState';
+import { OutputProps } from '../inputs/props';
+import { DefaultImageOutput } from '../outputs/DefaultImageOutput';
+import { DirectoryOutput } from '../outputs/DirectoryOutput';
 import { GenericOutput } from '../outputs/GenericOutput';
+import { LargeImageOutput } from '../outputs/LargeImageOutput';
+import { OutputContainer } from '../outputs/OutputContainer';
+import { TextOutput } from '../outputs/TextOutput';
 
-interface NodeOutputsProps {
-    id: string;
+interface FullOutputProps extends Omit<Omit<Output, 'type'>, 'id'>, OutputProps {
+    definitionType: Type;
+}
+
+// TODO: perhaps make this an object instead of a switch statement
+const pickOutput = (kind: OutputKind, props: FullOutputProps) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let OutputType: React.MemoExoticComponent<(props: any) => JSX.Element> = GenericOutput;
+    let isGenericType = true;
+    switch (kind) {
+        case 'image':
+            OutputType = DefaultImageOutput;
+            break;
+        case 'large-image':
+            OutputType = LargeImageOutput;
+            isGenericType = false;
+            break;
+        case 'directory':
+            OutputType = DirectoryOutput;
+            break;
+        case 'text':
+            OutputType = TextOutput;
+            break;
+        case 'generic':
+            return (
+                <OutputContainer
+                    generic
+                    definitionType={props.definitionType}
+                    hasHandle={props.hasHandle}
+                    id={props.id}
+                    key={`${props.id}-${props.outputId}`}
+                    label={props.label}
+                    outputId={props.outputId}
+                >
+                    <GenericOutput {...props} />
+                </OutputContainer>
+            );
```

Comment:
I don't think you need to treat this as anything special.

```suggestion
            OutputType = GenericOutput;
            break;
```